### PR TITLE
Improved XQuery and JSONiq support

### PR DESCRIPTION
**Pull request by wcandillon**
Hello,

I would like to have the syntax highlighting for the XQuery and JSONiq language based on these two Atom grammars: https://github.com/wcandillon/language-jsoniq/tree/tokenize/grammars.
Can someone point me into the right direction? :)

### CODE REVIEW

**Comment by pchaigno** [comment]
Please see instructions `2.` in [CONTRIBUTING.md#adding-a-language](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language) for how to add a grammar ;)

**Comment by wcandillon** [comment]
@pchaigno Thank you so much for this instructions. Can you review this please?
It is not sure to me why the build fails.

We're very exciting to see proper syntax highlighting for JSONiq and XQuery coming to github.

File: grammars.yml [hunk 1]
```diff
@@ -330,6 +330,9 @@ vendor/grammars/language-shellscript:
 - text.shell-session
 vendor/grammars/language-yaml:
 - source.yaml
+vendor/grammars/language-jsoniq
```
Comment (pchaigno) [inline grammars.yml]: There's a colon missing here (I'm surprised because you shouldn't have to edit this file manually).

**Comment by wcandillon** [comment]
@pchaigno Thanks a lot! I did edit them manually, I hope it's ok.

**Comment by pchaigno** [comment]
It won't work if you only edit the files manually. You need to follow the instructions to create the submodule ;)

**Comment by wcandillon** [comment]
@pchaigno I'm getting the following error while running the script
```bash
Williams-MacBook-Pro:linguist wcandillon$ script/convert-grammars --add vendor/grammars/language-jsoniq/
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- plist (LoadError)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from script/convert-grammars:6:in `<main>'
```
Does this look familiar?

**Comment by pchaigno** [comment]
I think you need to install the `plist` gem.

**Comment by wcandillon** [comment]
@pchaigno Thank you so much for your help, I proposed https://github.com/github/linguist/pull/2443 for merging. Exciting!

**Pull request closed without merging**